spec/api/comment_spec.rb

```diff
@@ -7,38 +7,48 @@
 
   describe "comments" do
     before(:each) { init_without_subscriptions }
+
     describe "GET /api/v1/comments/:comment_id" do
       it "returns JSON" do
-        comment = Comment.first
-        get "/api/v1/comments/#{comment.id}"
-        last_response.should be_ok
-        last_response.content_type.should == "application/json;charset=utf-8"
+        Comment.each do |comment|
+          get "/api/v1/comments/#{comment.id}"
+          last_response.should be_ok
+          last_response.content_type.should == "application/json;charset=utf-8"
+        end
       end
       it "retrieve information of a single comment" do
-        comment = Comment.first
-        get "/api/v1/comments/#{comment.id}"
-        last_response.should be_ok
-        retrieved = parse last_response.body
-        retrieved["body"].should == comment.body
-        retrieved["endorsed"].should == comment.endorsed
-        retrieved["id"].should == comment.id.to_s
-        retrieved["children"].should be_nil
-        retrieved["votes"]["point"].should == comment.votes_point
-        retrieved["depth"].should == comment.depth
-        retrieved["parent_id"].should == comment.parent_ids[-1]
+        Comment.each do |comment|
+          get "/api/v1/comments/#{comment.id}"
+          last_response.should be_ok
+          retrieved = parse last_response.body
+          retrieved["body"].should == comment.body
+          retrieved["endorsed"].should == comment.endorsed
+          retrieved["id"].should == comment.id.to_s
+          retrieved["children"].should be_nil
+          retrieved["votes"]["point"].should == comment.votes_point
+          retrieved["depth"].should == comment.depth
+          if comment.parent_ids.length > 0
+            retrieved["parent_id"].should == comment.parent_ids[-1].to_s
+          else
+            retrieved["parent_id"].should be_nil
+          end
+        end
       end
       it "retrieve information of a single comment with its sub comments" do
-        comment = Comment.first
-        get "/api/v1/comments/#{comment.id}", recursive: true
-        last_response.should be_ok
-        retrieved = parse last_response.body
-        retrieved["body"].should == comment.body
-        retrieved["endorsed"].should == comment.endorsed
-        retrieved["id"].should == comment.id.to_s
-        retrieved["votes"]["point"].should == comment.votes_point
-        retrieved["children"].length.should == comment.children.length
-        retrieved["children"].select{|c| c["body"] == comment.children.first.body}.first.should_not be_nil
-        retrieved["children"].each{|c| c["parent_id"].should == comment.id.to_s}
+        Comment.each do |comment|
```

Comment:
Same question: Why the change to validating each comment in the database vs only testing on a single comment?